pkg/util/ec2/imds_helpers.go

```diff
@@ -57,39 +66,50 @@ func getMetadataItemWithMaxLength(ctx context.Context, endpoint string, forceIMD
 	return result, err
 }
 
-func getMetadataItem(ctx context.Context, endpoint string, forceIMDSv2 bool) (string, error) {
+func getMetadataItem(ctx context.Context, endpoint string, action IMDSv2Usage) (string, error) {
 	if !pkgconfigsetup.IsCloudProviderEnabled(CloudProviderName, pkgconfigsetup.Datadog()) {
 		return "", fmt.Errorf("cloud provider is disabled by configuration")
 	}
 
-	return doHTTPRequest(ctx, metadataURL+endpoint, forceIMDSv2)
+	return doHTTPRequest(ctx, metadataURL+endpoint, action)
 }
 
 // UseIMDSv2 returns true if the agent should use IMDSv2
-func UseIMDSv2(forceIMDSv2 bool) bool {
-	return pkgconfigsetup.Datadog().GetBool("ec2_prefer_imdsv2") || forceIMDSv2
+func UseIMDSv2(force bool, disable bool) IMDSv2Usage {
```

Comment:
It's not intuitive to have 2 params to force using or not using V2.

What the point of calling `UseIMDSv2(true, false)` or `UseIMDSv2(false, true)` ? could we not simply replace those call by `true` and `false` and have this function only be use for case where we want to check the configuration ?

Reply:
> It's not intuitive to have 2 params to force using or not using V2.

I agree to this

> What the point of calling UseIMDSv2(true, false) or UseIMDSv2(false, true) ?

The idea was mainly to avoid the duplication of boolean to each function prototype, passing (..., force bool, disable bool) to each function it goes through was a bit heavy.

>could we not simply replace those call by true and false and have this function only be use for case where we want to check the configuration ?

Each time I call the function is because we want to check the configuration, thought about replacing directly with the IMDSv2 action to take but it needs to check the config to know if they need to use IMDSv2 or not.

Nonetheless I agree that's iffy, a solution may be to duplicate or refactor the code logic to achieve it in a more elegant way however it may need more times to do it.

WDYT ?

Reply:
done

Reply:
+1 on this function needing a bit of clean-up

Reply:
done in https://github.com/DataDog/datadog-agent/pull/30760